website/js/AlleleFrequencyCharts.js

```diff
@@ -5,141 +5,181 @@ var React = require('react'),
-            {label: 'NFE', prop: 'Allele_frequency_NFE_ExAC'},
-            {label: 'OTH', prop: 'Allele_frequency_OTH_ExAC'},
-            {label: 'SAS', prop: 'Allele_frequency_SAS_ExAC'},
-        ];
 
-        title = 'ExAC';
-        pointFormat =  "{point.y}<br /><em>({point.count} of {point.number})</em>";
-    } else {
-        return false;
-    }
+        if (state.scaleIndex !== nextState.scaleIndex) {
+            refs[`${props.prop}_alleleFreq2`].getChart().update({
+                yAxis: { max: this.scales[nextState.scaleIndex] }
+            });
+        }
 
-    if (_.every(_.values(_.pick(variant, _.map(frequencyProps, e => e.prop))), util.isEmptyField)) {
         return false;
     }
+    render() {
+        let {variant, prop} = this.props;
+        let frequencyProps, title, pointFormat;
+        if (prop === 'Allele_Frequency_Charts_1000_Genomes') {
+            if (!variant['Variant_in_1000_Genomes']) { // eslint-disable-line dot-notation
+                return false;
+            }
+            frequencyProps = [
+                {label: 'AFR', prop: 'AFR_Allele_frequency_1000_Genomes'},
+                {label: 'AMR', prop: 'AMR_Allele_frequency_1000_Genomes'},
+                {label: 'EAS', prop: 'EAS_Allele_frequency_1000_Genomes'},
+                {label: 'EUR', prop: 'EUR_Allele_frequency_1000_Genomes'},
+                {label: 'SAS', prop: 'SAS_Allele_frequency_1000_Genomes'},
+            ];
+            title = '1000 Genomes';
+            pointFormat =  "{point.y}";
+        }
+        else if (prop === 'Allele_Frequency_Charts_ExAC') {
+            if (!variant['Variant_in_ExAC']) {
+                return false;
+            }
+            frequencyProps = [
+                {label: 'AFR', prop: 'Allele_frequency_AFR_ExAC'},
+                {label: 'AMR', prop: 'Allele_frequency_AMR_ExAC'},
+                {label: 'EAS', prop: 'Allele_frequency_EAS_ExAC'},
+                {label: 'FIN', prop: 'Allele_frequency_FIN_ExAC'},
+                {label: 'NFE', prop: 'Allele_frequency_NFE_ExAC'},
+                {label: 'OTH', prop: 'Allele_frequency_OTH_ExAC'},
+                {label: 'SAS', prop: 'Allele_frequency_SAS_ExAC'},
+            ];
 
-    let categories = [];
-    let data = [];
+            title = 'ExAC';
+            pointFormat =  "{point.y}<br /><em>({point.count} of {point.number})</em>";
+        } else {
+            return false;
+        }
 
-    for (let frequencyType of frequencyProps) {
-        if (!util.isEmptyField(variant[frequencyType.prop])) {
-            categories.push(frequencyType.label);
-            data.push({
-                y: parseFloat(variant[frequencyType.prop]),
-                number: variant[frequencyType.prop.replace("frequency", "number")],
-                count: variant[frequencyType.prop.replace("frequency", "count")]
-            });
+        if (_.every(_.values(_.pick(variant, _.map(frequencyProps, e => e.prop))), util.isEmptyField)) {
```

Comment:
could you add a comment explanation here to save future devs from trying to read this?

Reply:
I also think i've seen this before, if so it could be wrapped into a function with an explanation